transformer/classes/dockerfile/rubydockerfilegenerator.go

```diff
@@ -0,0 +1,165 @@
+				serviceName: []transformertypes.TransformerPlan{{
+					Mode:              t.Config.Spec.Mode,
+					ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+					BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+					Paths: map[string][]string{
+						artifacts.ProjectPathPathType: {dir},
+					},
+				}},
+			}
+			return namedServices, nil, nil
+		}
+		unnamedServices = []transformertypes.TransformerPlan{{
+			Mode:              t.Config.Spec.Mode,
+			ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			Paths: map[string][]string{
+				artifacts.ProjectPathPathType: {dir},
+			},
+		}}
+		return nil, unnamedServices, nil
+	}
+	return nil, nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *RubyDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var rubyConfig RubyTemplateConfig
+		var detectedPorts []int32
+		detectedPorts = append(detectedPorts, 8080) //TODO: Write parser to parse and identify port
+		rubyConfig.Port = commonqa.GetPortForService(detectedPorts, a.Name)
+		rubyConfig.AppName = a.Name
+		if sImageName.ImageName == "" {
+			sImageName.ImageName = common.MakeStringContainerImageNameCompliant(sConfig.ServiceName)
+		}
+		pathMappings = append(pathMappings, transformertypes.PathMapping{
+			Type:     transformertypes.SourcePathMappingType,
+			DestPath: common.DefaultSourceDir,
+		}, transformertypes.PathMapping{
+			Type:           transformertypes.TemplatePathMappingType,
+			SrcPath:        filepath.Join(t.Env.Context, t.Config.Spec.TemplatesDir),
+			DestPath:       filepath.Join(common.DefaultSourceDir, relSrcPath),
+			TemplateConfig: rubyConfig,
+		})
+		paths := a.Paths
+		paths[artifacts.DockerfilePathType] = []string{filepath.Join(common.DefaultSourceDir, relSrcPath, "Dockerfile")}
```

Comment:
I think we have a constant for Dockerfile string.

Reply:
Yes, I have changed the hard-coded string to the Dockerfile string.